jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/core/threadgroups/DslDefaultThreadGroup.java

```diff
@@ -283,15 +283,19 @@ public DslDefaultThreadGroup holdIterating(String iterations) {
     addStage(new Stage(getLastStage().threadCount(), null, iterations));
     return this;
   }
-
   private void checkIterationsPreConditions() {
-    if (!(stages.size() == 1 && !ZERO.equals(stages.get(0).threadCount())
-        || stages.size() == 2 && ZERO.equals(stages.get(0).threadCount())
-        && !ZERO.equals(stages.get(1).threadCount()))) {
+    boolean isSingleStageHold = stages.size() == 1 && !ZERO.equals(stages.get(0).threadCount());
```

Comment:
I like the spirit of this refactor! But what do you think of something like this?:

```java
boolean isJustRamp = stages.size() == 1 && !ZERO.equals(stages.get(0).threadCount());
boolean isJustDelayAndRamp = stages.size() == 2 && ZERO.equals(stages.get(0).threadCount())
        && !ZERO.equals(stages.get(1).threadCount()))
if (!(isJustRamp || isJustDelayAndRamp)) {
  throw new IllegalStateException("Holding for iterations is only supported after initial hold and ramp, or ramp.");
}
if (ZERO.equals(getLastStage().threadCount())) {
....
```

I think adding a variables for each component of the or is a good idea since the condition of the if is quite big otherwise. Then for the two remaining if conditions (negated or and comparing last stage thread count to zero) I think adding variables (and the additional code lines) is not worth it, since they are pretty simple and understandable.